vector/src/Data/Vector/Generic.hs

```diff
@@ -244,10 +244,21 @@ v !? i | i `inRange` length v = case basicUnsafeIndexM v i of Box a -> Just a
 
 
 -- | /O(1)/ First element.
+--
+-- Partial.
+-- Undefined at @fromList [ ]@.
+-- Completion is 'vectorToMaybe'
 head :: Vector v a => v a -> a
 {-# INLINE_FUSED head #-}
 head v = v ! 0
 
+-- | /O(1)/ Just the first element — or nothing when there are none.
+--
+-- Completes 'head'.
+vectorToMaybe :: Vector v a => v a -> Maybe a
+{-# INLINE_FUSED vectorToMaybe #-}
+vectorToMaybe v = v !? 0
+
```

Comment:
Such function needs some haddock and needs to be replicated through all of the variants (Storable, Primitive, Unboxed and boxed)

Reply:
But there is some haddock. Do you have something specific in mind?

We shall replicate it once the design is clear. Now is not yet time since even its name may change.

Reply:
Adding doctest is a general suggestion, not a requirement. Checkout examples for some of the other functions. 

> We shall replicate it in the way you indicate once the design is clear. Now is not yet time since even its name may change.

Absolutely. I was just making suggestions on the rest of the things that need to be done on the PR

Reply:
I am confused. First you said:

> Such function needs some haddock …

— Then you said:

> Adding doctest is a general suggestion, not a requirement.

— Where did `doctest` come from? Do you want me to add some examples in the `>>>` style?

Reply:
> … needs to be replicated through all of the variants …

I have edited #479 accordingly.

Reply:
> Such function needs some haddock 

I am really sorry! I really meant to say doctest. You do have haddock for the function, so I can totally see how my comment didn't make any sense.

Reply:
I see. No problem.

I do not see a lot of `doctest` examples right now. For example, `head` does not have one. The first occurrence of the string `>>>` is at line 894 in `Data.Vector.Generic`. Would you suggest adding more examples to the code base overall? _(Not necessarily in this pull request, but rather in principle.)_

Reply:
There are not many doctests in Generic, because a doctest requires concrete types for it to work. So, all of the doctests are in modules for concrete vectors: Unboxed, Storable, etc

Reply:
I see. When we get there, we shall do that.